espoon-voltti/hoivakoti-app
# feat/HOIV-130-auth-container

#### Summary

- Adding Keycloak as a identity provider for Valvonta pages
- Add backend routes and controllers for authentication
- Add Nginx container as a proxy
- Protect PageReportsAdmin and PageOpenFeedbackResults components with Authentication context

#### Dependencies

Infra [PR](https://github.com/espoon-voltti/hoivakoti-infra/pull/3)

#### Testing instructions

Run docker by including both docker-composey.yml and docker-compose.keycloak.yml files. Keycloak realm should be imported from `keycloak/hoivakodit-realm.json`. Navigate to `/auth/admin`and login with admin credentials (password same as one used for admin tools). In Keycloak console create yourself and test user and add it to "Valvonta Team" group to give yourself access. Try logging in and using `valvonta` and `valvonta/palaute` pages.

#### Checklist for pull request creator


- [x] A task has been created for the PR on the Kanban board with necessary details filled (one task / repo)+t#Commit-viestien-muotoilu)
- [ ] Tests have been written for the change
- [x] All tests pass (unit, e2e)
- [x] All new code has been linted and there aren't any lint errors
- [x] The change has been tested locally, e.g., with httpie
- [x] The change has been tested in the browser with the latest Chrome
- [x] The change has been tested with a smaller screen (tablet and smartphone / emulator)
- [ ] All translations have been added (fi and sv)
- [x] The code is self-documenting or has been documented sufficiently, e.g., in the README or inline
- [x] The branch has been rebased against dev branch before the PR was created

#### Checklist for pull request reviewer (copy to review text box)


```
- [ ] A task has been created for the PR on the Kanban board with necessary details filled (one task / repo)
- [ ] Tests have been written for the change
- [ ] All tests pass (unit, e2e)
- [ ] All new code has been linted and there aren't any lint errors
- [ ] The change has been tested locally, e.g., with httpie
- [ ] The change has been tested in the browser with the latest Chrome
- [ ] The change has been tested with a smaller screen (tablet and smartphone / emulator)
- [ ] All translations have been added (fi and sv)
- [ ] The code is self-documenting or has been documented sufficiently, e.g., in the README or inline
- [ ] The PR branch has been rebased against dev branch and force pushed if necessary before merging
```
